src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Models/ProcessContext.cs

```diff
@@ -1,10 +1,10 @@
 ﻿using System.Collections.Generic;
-using Hl7.Fhir.ElementModel;
 
 namespace Microsoft.Health.Fhir.Anonymizer.Core.Models
 {
     public class ProcessContext
     {
-        public HashSet<ElementNode> VisitedNodes { get; set; }
+        // The location of visited nodes, e.g., Patient.name[0].use[0]
+        public HashSet<string> VisitedNodes { get; set; }
```

Comment:
Can we use HashSet<ITypedElement> here?

Reply:
If we use `HashSet<ITypedElement>` to check whether an `ElementNode` is in it, it will always return false.
```csharp
var rootNode = FhirJsonNode.Parse(inputFileContent).ToTypedElement(provider);
var visitedNodes = new HashSet<ITypedElement> { rootNode };
var result = visitedNodes.Contains(ElementNode.FromElement(rootNode)); // false
```